rubix-prestosql/src/main/java/com/qubole/rubix/prestosql/PrestoClusterManager.java

```diff
@@ -12,194 +12,41 @@
  */
 package com.qubole.rubix.prestosql;
 
-import com.google.common.base.Throwables;
-import com.google.common.cache.CacheBuilder;
-import com.google.common.cache.CacheLoader;
-import com.google.common.cache.LoadingCache;
 import com.google.common.collect.ImmutableList;
-import com.google.common.collect.Lists;
-import com.google.common.reflect.TypeToken;
-import com.google.gson.Gson;
-import com.qubole.rubix.common.utils.ClusterUtil;
 import com.qubole.rubix.spi.ClusterManager;
 import com.qubole.rubix.spi.ClusterType;
-import org.apache.commons.logging.Log;
-import org.apache.commons.logging.LogFactory;
-import org.apache.hadoop.conf.Configuration;
+import io.prestosql.spi.Node;
+import io.prestosql.spi.NodeManager;
 
-import java.io.BufferedReader;
-import java.io.IOException;
-import java.io.InputStreamReader;
-import java.lang.reflect.Type;
-import java.net.HttpURLConnection;
-import java.net.InetAddress;
-import java.net.MalformedURLException;
-import java.net.URI;
-import java.net.URL;
-import java.util.Collections;
-import java.util.HashSet;
 import java.util.List;
-import java.util.Objects;
-import java.util.Set;
-import java.util.concurrent.ExecutionException;
-import java.util.concurrent.ExecutorService;
-import java.util.concurrent.Executors;
-import java.util.concurrent.TimeUnit;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.qubole.rubix.spi.ClusterType.PRESTOSQL_CLUSTER_MANAGER;
+import static java.util.Objects.requireNonNull;
 
 /**
  * Created by stagra on 14/1/16.
  */
 public class PrestoClusterManager extends ClusterManager
 {
-  private boolean isMaster = true;
-  private int serverPort = 8081;
-  private String serverAddress = "localhost";
-  static LoadingCache<String, List<String>> nodesCache;
-
-  private Log log = LogFactory.getLog(PrestoClusterManager.class);
+  private static volatile NodeManager nodeManager;
 
-  public static String serverPortConf = "caching.fs.presto-server-port";
+  public static void setNodeManager(NodeManager nodeManager)
+  {
+    PrestoClusterManager.nodeManager = requireNonNull(nodeManager, "nodeManager is null");
+  }
 
-  // Safe to use single instance of HttpClient since Supplier.get() provides synchronization
   @Override
-  public void initialize(Configuration conf)
+  public ClusterType getClusterType()
```

Comment:
We need to retain the existing code for standalone mode. Defining new cluster manager for embedded mode would require many changes, it would be simpler when we decouple FileSystems and ClusterManagers. For now, we need to have two suppliers of nodeList: native NodeManager based and /v1/node based, with native one, when set, getting preference over the other.

Reply:
> We need to retain the existing code for standalone mode.

Doesn't Rubix manage nodes themselves in standalone mode? Why `PrestoClusterManager` is needed then?